curator-framework/src/main/java/org/apache/curator/framework/imps/CreateBuilderImpl.java

```diff
@@ -1212,7 +1212,21 @@ public String call() throws Exception
                             {
                                 if ( setDataIfExists )
                                 {
-                                    client.getZooKeeper().setData(path, data, setDataIfExistsVersion);
+                                    Stat setStat = client.getZooKeeper().setData(path, data, setDataIfExistsVersion);
+                                    if(storingStat != null)
+                                    {
+                                        storingStat.setAversion(setStat.getAversion());
```

Comment:
You can use `DataTree#copyStat()` - FYI

Reply:
Other than that LGTM

Reply:
Thanks, there was another place in CreateBuilderImpl doing the same thing, and I've replaced that with DataTree#copyStat() also.